### PR TITLE
MEGATRON-8341: stop Auto sync job on keycloak auth server

### PR DESCRIPTION
**Pull request by AnkurJaiswal1990**

### CODE REVIEW

File: federation/ldap/src/main/java/org/keycloak/storage/ldap/LDAPStorageProviderFactory.java [hunk 1]
```diff
@@ -469,7 +469,11 @@ public void preRemove(KeycloakSession session, RealmModel realm, ComponentModel
 
     @Override
     public SynchronizationResult sync(KeycloakSessionFactory sessionFactory, String realmId, UserStorageProviderModel model) {
-        syncMappers(sessionFactory, realmId, model);
+    	if(Boolean.FALSE.equals(Boolean.parseBoolean(System.getenv("ENABLED_AUTO_SYNC_JOB")))) {
```
Comment (thakkarnirav-duclo) [inline federation/ldap/src/main/java/org/keycloak/storage/ldap/LDAPStorageProviderFactory.java]: @AnkurJaiswal1990 why boolean compare with equal here?

File: federation/ldap/src/main/java/org/keycloak/storage/ldap/LDAPStorageProviderFactory.java [hunk 2]
```diff
@@ -485,7 +489,11 @@ public SynchronizationResult sync(KeycloakSessionFactory sessionFactory, String
 
     @Override
     public SynchronizationResult syncSince(Date lastSync, KeycloakSessionFactory sessionFactory, String realmId, UserStorageProviderModel model) {
-        syncMappers(sessionFactory, realmId, model);
+    	if(Boolean.FALSE.equals(Boolean.parseBoolean(System.getenv("ENABLED_AUTO_SYNC_JOB")))) {
```
Comment (thakkarnirav-duclo) [inline federation/ldap/src/main/java/org/keycloak/storage/ldap/LDAPStorageProviderFactory.java]: @AnkurJaiswal1990 why boolean compare with equal here?

File: federation/ldap/src/main/java/org/keycloak/storage/ldap/LDAPStorageProviderFactory.java (hunk 1 above)
Comment (AnkurJaiswal1990) [reply federation/ldap/src/main/java/org/keycloak/storage/ldap/LDAPStorageProviderFactory.java]: @thakkarnirav-duclo done

File: federation/ldap/src/main/java/org/keycloak/storage/ldap/LDAPStorageProviderFactory.java (hunk 2 above)
Comment (AnkurJaiswal1990) [reply federation/ldap/src/main/java/org/keycloak/storage/ldap/LDAPStorageProviderFactory.java]: @thakkarnirav-duclo done

File: federation/ldap/src/main/java/org/keycloak/storage/ldap/LDAPStorageProviderFactory.java [hunk 3]
```diff
@@ -469,7 +469,11 @@ public void preRemove(KeycloakSession session, RealmModel realm, ComponentModel
 
     @Override
     public SynchronizationResult sync(KeycloakSessionFactory sessionFactory, String realmId, UserStorageProviderModel model) {
-        syncMappers(sessionFactory, realmId, model);
+    	if(false == (Boolean.parseBoolean(System.getenv("ENABLED_AUTO_SYNC_JOB")))) {
```
Comment (thakkarnirav-duclo) [inline federation/ldap/src/main/java/org/keycloak/storage/ldap/LDAPStorageProviderFactory.java]: @AnkurJaiswal1990 

if (Boolean) {
}

**Review by thakkarnirav-duclo — CHANGES_REQUESTED** [review]
@AnkurJaiswal1990  boolean  condition has to be like if(boolean), you should not required to compare with equals or == 

CC: @amitadalja

File: federation/ldap/src/main/java/org/keycloak/storage/ldap/LDAPStorageProviderFactory.java (hunk 3 above)
Comment (AnkurJaiswal1990) [reply federation/ldap/src/main/java/org/keycloak/storage/ldap/LDAPStorageProviderFactory.java]: Hello @thakkarnirav-duclo done as per review comment.

**Review by thakkarnirav-duclo — APPROVED** [review]

**Pull request merged**